core/trino-main/src/main/java/io/trino/operator/scalar/ArrayRemoveFunction.java

```diff
@@ -56,12 +55,12 @@ public Block remove(
                     operator = EQUAL,
                     argumentTypes = {"E", "E"},
                     convention = @Convention(arguments = {NEVER_NULL, NEVER_NULL}, result = NULLABLE_RETURN))
-                    MethodHandle equalsFunction,
+                    MethodHandle equalFunction,
             @TypeParameter("E") Type type,
             @SqlType("array(E)") Block array,
             @SqlType("E") long value)
     {
-        return remove(equalsFunction, type, array, (Object) value);
+        return remove(equalFunction, type, array, (Object) value);
```

Comment:
Are all of these methods still needed? Can we convert to `Object` automatically?